contracts/Removal.sol

```diff
@@ -284,6 +295,32 @@ contract Removal is
     super.safeBatchTransferFrom(from, to, ids, amounts, projectId);
   }
 
+  /**
+   * @notice Marks an amount of a removal as released given a removal ID and a quantity.
+   * @param removalId The ID of the removal to mark as released.
+   * @param amount The amount of the removal to release.
+   *
+   * ##### Requirements:
+   *
+   * - The contract must not be paused.
+   * - The caller must have the `RELEASER_ROLE`.
+   * - The `amount` for the removal must be <= the removal's initial amount
+   */
+  function release(
+    address owner,
+    uint256 removalId,
+    uint256 amount
+  ) external whenNotPaused onlyRole(RELEASER_ROLE) {
+    // todo initialBalanceOf? should be amount + released?
+    // Querying the details of a removal returns a flag showing the quantity that was invalidated.
+    // If this Removal has not been completely sold, we will not sell the invalidated amount of that Removal
+    _tokenIdToReleased[removalId] = amount;
+    burn(owner, removalId, amount);
```

Comment:
Not sure I follow

Why wouldn't we know how much to burn/release?

Reply:
so the issue is that `amount` represents the total amount of that removal's released carbon, regardless of whether that carbon has already been sold or not (and we use the mapping to store that information for accountability and transparency).  However, this total released value isn't necessarily the same amount that should be getting burned because it's possible that the removal has been totally or partially sold through, and that there isn't enough remaining balance on the removal to burn the full released amount (this represents the case where certificates have to be amended)

Reply:
@amiecorso is this still an issue in the current commit?

Reply:
discussed on a call and you're addressing in #305